### PR TITLE
Optionally use openssl instead of boring ssl.

### PR DESCRIPTION
**Pull request by bacek**
This prevent link/runtime failure when you have a crate which uses both
grpc-rs and openssl-rs.

Based on feedback from #186. Build against fresh master.

### CODE REVIEW

File: grpc-sys/Cargo.toml [hunk 1]
```diff
@@ -47,10 +47,12 @@ exclude = [
 
 [dependencies]
 libc = "0.2"
+# openssl = { version = "*", optional = true }
```
Comment (overvenus) [inline grpc-sys/Cargo.toml]: It can be removed.

**Comment by overvenus** [comment]
Cmake can not find openssl on macOS. https://travis-ci.org/pingcap/grpc-rs/jobs/439426196#L1504

```
--- stderr
CMake Error at /usr/local/Cellar/cmake/3.11.4/share/cmake/Modules/FindPackageHandleStandardArgs.cmake:137 (message):
  Could NOT find OpenSSL, try to set the path to OpenSSL root folder in the
  system variable OPENSSL_ROOT_DIR (missing: OPENSSL_INCLUDE_DIR)
Call Stack (most recent call first):
  /usr/local/Cellar/cmake/3.11.4/share/cmake/Modules/FindPackageHandleStandardArgs.cmake:378 (_FPHSA_FAILURE_MESSAGE)
  /usr/local/Cellar/cmake/3.11.4/share/cmake/Modules/FindOpenSSL.cmake:390 (find_package_handle_standard_args)
  cmake/ssl.cmake:38 (find_package)
  CMakeLists.txt:119 (include)
```

File: grpc-sys/Cargo.toml (hunk 1 above)
Comment (bacek) [reply grpc-sys/Cargo.toml]: Ooops. Removed.

**Comment by bacek** [comment]
OSX build are fixed now. rust-beta are failing independently from openssl.

**Comment by bacek** [comment]
@overvenus Can you merge it? Looks like automatic merge is blocked by rust-beta Travis failure.

**Comment by overvenus** [comment]
I can rerun travis CI, but still need one approval from maintainer in order to merge.

PTAL @BusyJay

**Comment by bacek** [comment]
Is RemoteStopped legitimate response in case of dropping tx/rx channel? Because looks like there is a race condition between Sink::start_send and async::check_alive.

**Pull request merged**